desktop/apps/consignments2/client/actions.js

```diff
@@ -21,32 +26,107 @@ export function chooseArtistAndAdvance (value) {
   }
 }
 
+export function chooseLocation (location) {
+  return async (dispatch, getState) => {
+    const parseDetails = (place, status) => {
+      if (status !== window.google.maps.places.PlacesServiceStatus.OK) {
+        console.error('Unable to reach maps API', status)
+      } else {
+        const { address_components } = place
+        const city = find(address_components, (comp) => { return comp.types[0] === 'locality' })
+        const state = find(address_components, (comp) => { return comp.types[0] === 'administrative_area_level_1' })
+        const country = find(address_components, (comp) => { return comp.types[0] === 'country' })
+
+        const cityDisplay = city && city.long_name
+        const countryDisplay = country && country.long_name
+        const stateDisplay = state && state.long_name
+
+        dispatch(updateLocationInputValues(cityDisplay, stateDisplay, countryDisplay))
+      }
+    }
+
+    try {
+      if (window.google) {
+        const placesService = new window.google.maps.places.PlacesService(document.createElement('div'))
```

Comment:
oh ugh. yeah you have to either pass in a full-fledged `map` object there, or a dummy node on the page 😞